website/src/content/development/forms.mdx

```diff
@@ -0,0 +1,673 @@
+            label="Cancel"
+            onClick={formik.handleReset}
+          />
+          <PrimaryButton
+            type="submit"
+            label="Submit"
+            onClick={formik.handleSubmit}
+            isDisabled={formik.isSubmitting}
+          />
+        </Spacings.Inline>
+      </Spacings.Stack>
+    </form>
+  );
+}
+```
+
+Now that we have defined our form component, we can implement the create and details pages.
+
+The create page does not have any initial data, so we can use our conversion function `docToFormValues()` with default values.
+
+```js title="channels-create.js"
+import { useCallback } from 'react';
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+import Text from '@commercetools-uikit/text';
+import Spacings from '@commercetools-uikit/spacings';
+import { docToFormValues, formValuesToDoc } from './conversions';
+import ChannelsForm from './channels-form';
+
+const ChannelsCreate = () => {
+  const languages = useApplicationContext((context) => context.project.languages);
+  const handleSubmit = useCallback(
+    async (formValues) => {
+      const data = formValuesToDoc(formValues);
+      // This would trigger the request, for example a mutation.
+      const result = await createChannel(data);
+      // If successful, show a notification and redirect
+      // to the Channels details page.
+      // If errored, show an error notification.
+    },
+    [createChannel]
+  );
+
+  return (
+    <Spacings.Stack scale="xl">
+      <Text.Headline as="h1">
+        Create a channel
+      </Text.Headline>
+      <ChannelsForm
+        initialValues={docToFormValues(null, languages)}
+        onSubmit={handleSubmit}
+      />
+    </Spacings.Stack>
+  );
+}
+```
+
+In the details page we would need to fetch the data first, then initialize the form with the data.
+
+```js title="channels-details.js"
+import { useCallback } from 'react';
+import { useRouteMatch } from "react-router-dom";
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+import Text from '@commercetools-uikit/text';
+import Spacings from '@commercetools-uikit/spacings';
+import LoadingSpinner from '@commercetools-uikit/loading-spinner';
+import useChannels from './use-channels';
+import { docToFormValues, formValuesToDoc } from './conversions';
+import ChannelsForm from './channels-form';
+
+const ChannelsDetails = (props) => {
+  const match = useRouteMatch();
+  const languages = useApplicationContext((context) => context.project.languages);
+  const { data: channel } = useChannels(match.params.id)
+  const handleSubmit = useCallback(
+    async (formValues) => {
+      const data = formValuesToDoc(formValues);
+      // This would trigger the request, for example a mutation.
+      const result = await updateChannel(data);
+      // If successful, show a notification.
+      // If errored, show an error notification.
+    },
+    [updateChannel]
+  );
```

Comment:
Since this is a detailed example and all the dependencies are defined, I'm missing where this `updateChannel` function comes from.